frontend/awx/analytics/useAnalyticsReportBuilderView.ts

```diff
@@ -0,0 +1,246 @@
+
+  // If a column is defined with defaultSort:true use that column to set the default sort, otherwise use the first column
+  if (tableColumns && tableColumns.length) {
+    const defaultSortColumn = tableColumns.find((column) => column.defaultSort) ?? tableColumns[0];
+    defaultSort = defaultSortColumn?.sort;
+    defaultSortDirection = defaultSortColumn?.defaultSortDirection;
+  }
+
+  const view = useView({
+    defaultValues: {
+      sort: defaultSort,
+      sortDirection: defaultSortDirection,
+      filterState: defaultFilters,
+    },
+    disableQueryString,
+    ignoreQueryStringKeys: ['reportName', 'chart_type'],
+  });
+
+  const { page, perPage, sort, sortDirection, filterState } = view;
+
+  let queryString = queryParams ? `?${getQueryString(queryParams)}` : '';
+
+  if (sort) {
+    let canSort = tableColumns?.find((item) => item.sort == sort) ? true : false;
+
+    if (!canSort) {
+      // try sortableColumns instead if table columns not defined
+      canSort = sortableColumns?.find((item) => item == sort) ? true : false;
+    }
+
+    if (canSort) {
+      if (!sortKey) {
+        sortKey = 'sort_by';
+      }
+      queryString ? (queryString += '&') : (queryString += '?');
+      if (sortDirection === 'desc') {
+        queryString += `${sortKey}=${sort}:desc`;
+      } else {
+        queryString += `${sortKey}=${sort}:asc`;
+      }
+    }
+  }
+
+  fillFilters(postData, filterState, toolbarFilters || []);
+
+  queryString ? (queryString += '&') : (queryString += '?');
+  queryString += `offset=${(page - 1) * perPage}`;
+
+  queryString ? (queryString += '&') : (queryString += '?');
+  queryString += `limit=${perPage}`;
+
+  url += queryString;
+
+  // we cant only run fetch when url changes, but also when something in postData changes
+  const postDataJson = JSON.stringify(postData);
+  const changed = url + postDataJson;
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const data = await postRequest(url, postData);
+        setData(data);
+      } catch (error) {
+        setError(error);
+      }
+    }
+
+    if (!disableLoading) {
+      void (async () => {
+        await fetchData();
+      })();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [changed, disableLoading]);
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  if (error instanceof RequestError) {
+    if (error.statusCode === 404 && view.page > 1) {
+      view.setPage(1);
+      setError(undefined);
+    }
+  }
+
+  const selection = useSelected(data?.meta.legend ?? [], keyFn, defaultSelection);
+
+  return useMemo(() => {
+    return {
+      itemCount: data?.meta.count,
+      pageItems: data?.meta.legend,
+      error,
+      ...view,
+      ...selection,
+      originalData: data as undefined,
+    };
+  }, [error, selection, view, data]);
+}
+
+export function getAwxError(err: unknown) {
```

Comment:
In the follow-up PR we need to introduced our own error handling because backend is sending back info on what is set wrongly and we need to show specific error message with hint how to fix it.